chemprop/hyperopt_utils.py

```diff
@@ -5,11 +5,43 @@
 import csv
 import json
 
-from hyperopt import Trials
+from hyperopt import Trials, hp
+import numpy as np
 
 from chemprop.constants import HYPEROPT_SEED_FILE_NAME
 from chemprop.utils import makedirs
 
+def build_search_space(search_parameters: List[str], train_epochs: int = None) -> dict:
+    """
+    Builds the parameter space to be searched with hyperopt trials.
+
+    :param search_parameters: A list of parameters to be included in the search space.
+    :param train_epochs: The total number of epochs to be used in training.
+    :return: A dictionary keyed by the parameter names of hyperopt search functions.
+    """
+    available_spaces = {
+        "activation": hp.choice("activation", options=['ReLU', 'LeakyReLU', 'PReLU', 'tanh', 'SELU', 'ELU']),
+        "aggregation": hp.choice("aggregation", options=["mean", "sum", "norm"]),
+        "aggregation_norm": hp.quniform("aggregation_norm", low=1, high=200, q=1),
+        "batch_size": hp.quniform("batch_size", low=10, high=200, q=10),
+        "depth": hp.quniform("depth", low=2, high=6, q=1),
+        "dropout": hp.quniform("dropout", low=0.0, high=0.4, q=0.05),
+        "ffn_hidden_size": hp.quniform("ffn_hidden_size", low=300, high=2400, q=100),
+        "ffn_num_layers": hp.quniform("ffn_num_layers", low=1, high=3, q=1),
+        "final_lr_ratio": hp.loguniform("final_lr_ratio", low=np.log(1e-4), high=0.),
+        "hidden_size": hp.quniform("hidden_size", low=300, high=2400, q=100),
+        "init_lr_ratio": hp.loguniform("init_lr_ratio", low=np.log(1e-4), high=0.),
+        "linked_hidden_size": hp.quniform("linked_hidden_size", low=300, high=2400, q=100),
+        "max_lr": hp.loguniform("max_lr", low=np.log(1e-6), high=np.log(1e-2)),
+        "warmup_epochs": hp.quniform("warmup_epochs", low=1, high=train_epochs // 2, q=1)
+    }
```

Comment:
Since the searching range of each parameter is hard coded here, users might want to change this range. For example, if `--features_only` is used, only the additional features will be passed to FFN, no graph network, and it doesn't make sense to still search the hyperparameter of `ffn_hidden_size` starting from 300 to 2400. I think it would be better to add some description like "The searching range of each hyperparams has been hard coded, and if you want to change it, please change the object of available_spaces in build_search_space() in  hyperopt_utils.py". I think this change is not necessary to be made since the default setting is generally enough for most cases.

Reply:
I think this would be useful in the readme and have added it.